src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -260,6 +265,23 @@ public Task Completion
         /// <value>The default value is <see cref="Formatting.Indented"/>.</value>
         public Formatting JsonSerializerFormatting { get; set; } = Formatting.Indented;
 
+        /// <summary>
+        /// Gets or sets a value indicating whether to cancel all methods dispatched locally
+        /// that accept a <see cref="CancellationToken"/> when the connection with the remote party is closed.
+        /// </summary>
+        public bool CancelLocallyInvokedMethodsWhenConnectionIsClosed
+        {
+            get => this.cancelLocallyInvokedMethodsWhenConnectionIsClosed;
+            set
+            {
+                // We don't typically allow changing this setting after listening has started because
```

Comment:
What cases do we allow it? Seems like we always throw if we've started listening

Reply:
The method we're calling here does not throw if `AllowModificationWhileListening` has been set to true.

https://github.com/Microsoft/vs-streamjsonrpc/blob/cffa2677a9f5fce140e086c43f2446efdd61f7ff/src/StreamJsonRpc/JsonRpc.cs#L1494-L1497